package.json

```diff
@@ -20,38 +20,60 @@
     "@semantic-release/error": "^2.0.0",
     "@semantic-release/last-release-npm": "^2.0.0",
     "@semantic-release/release-notes-generator": "^4.0.0",
+    "execa": "^0.8.0",
+    "fs-extra": "^4.0.2",
     "git-head": "^1.2.1",
-    "github": "^8.0.0",
+    "github": "^11.0.0",
     "lodash": "^4.0.0",
     "nerf-dart": "^1.0.0",
     "nopt": "^4.0.0",
     "normalize-package-data": "^2.3.4",
     "npmconf": "^2.1.2",
     "npmlog": "^4.0.0",
+    "p-series": "^1.0.0",
     "parse-github-repo-url": "^1.3.0",
     "require-relative": "^0.8.7",
-    "run-auto": "^2.0.0",
-    "run-series": "^1.1.3",
-    "semver": "^5.2.0"
+    "semver": "^5.4.1"
   },
   "devDependencies": {
-    "coveralls": "^3.0.0",
+    "ava": "^0.22.0",
+    "codecov": "^2.3.0",
+    "commitizen": "^2.9.6",
     "cz-conventional-changelog": "^2.0.0",
-    "mkdirp": "^0.5.1",
-    "mock-spawn": "^0.2.6",
-    "nixt": "^0.5.0",
+    "eslint": "^4.7.0",
+    "eslint-config-prettier": "^2.5.0",
+    "eslint-config-standard": "^10.2.1",
+    "eslint-plugin-import": "^2.7.0",
+    "eslint-plugin-node": "^5.2.0",
+    "eslint-plugin-prettier": "^2.3.0",
+    "eslint-plugin-promise": "^3.5.0",
+    "eslint-plugin-standard": "^3.0.1",
     "nock": "^9.0.2",
     "npm-registry-couchapp": "^2.6.12",
-    "nyc": "^10.0.0",
+    "nyc": "^11.2.1",
+    "p-map-series": "^1.0.0",
+    "prettier": "^1.7.0",
```

Comment:
Isn't it better to integrate prettier + eslint in `pre-commit hook`? 
So we can make sure prettier + eslint reformat any changes before committing

Reply:
Also this approach will reduce the possibility of us having code style discussion in code review
https://github.com/semantic-release/semantic-release/pull/451/files#r141981878

Reply:
We had this discussion about commit hook previously among maintainer and we've decided to not use commit hook for simplicity. Also that could create confusion for contributors to see their code automatically changed on commit.

The chosen solution is to enforce the formating with eslint (with eslint-plugin-prettier).
All the code format pushed is validated in CI with eslint to be sure it's formatted with prettier.

This [comment](https://github.com/semantic-release/semantic-release/pull/451/files#r141981878) is not really about style but more about a not well know javascript feature allowing to reassign `let` variables with destructuring.

Reply:
> We had this discussion about commit hook previously among maintainer...

cool, I respect that 👍 

> The chosen solution is to enforce the formating with eslint (with eslint-plugin-prettier).
> All the code format pushed is validated in CI with eslint to be sure it's formatted with prettier.

yeah, the changes will failed on CI if they're not following our prettier config but then we still have to discuss about the code style things in the PR, right? isn't it not so good in terms of productivity?

Reply:
Not really because `eslint` will run on `npm run test`. So a contributor would figure that out before committing.
Sure we could run `npm run test` on each commit or something like that but:
- Most contributor would know they have to at least run unit test before commiting
- For some less experienced contributors, we prefer answer them in the PR and help them as we believe guidance from a human would be more pleasant and helpful than an error message on commit

And there wouldn't be any style discussion, the style is enforced by eslint + prettier. There would be just explanation on how to read the eslint error, how to format properly, and maybe suggestion to use tools etc...Anything helpful for the contributor.

Reply:
they will also fail when you run `npm test` locally